azure-mgmt-keyvault/src/main/java/com/microsoft/azure/management/keyvault/Key.java

```diff
@@ -0,0 +1,457 @@
+
+    /**
+     * Container interface for all the definitions.
+     */
+    interface Definition extends
+            DefinitionStages.Blank,
+            WithKey,
+            DefinitionStages.WithImport,
+            DefinitionStages.WithCreate {
+    }
+
+    /**
+     * Grouping of key definition stages.
+     */
+    interface DefinitionStages {
+        /**
+         * The first stage of a key definition.
+         */
+        interface Blank extends WithKey {
+        }
+
+        /**
+         * The stage of a key definition allowing to specify whether
+         * to create a key or to import a key.
+         */
+        interface WithKey {
+            /**
+             * Specifies a key type to create a new key.
+             * @param keyType the JWK type to create
+             * @return the next stage of the definition
+             */
+            WithCreate withKeyTypeToCreate(JsonWebKeyType keyType);
+
+            /**
+             * Specifies an existing key to import.
+             * @param key the existing JWK to import
+             * @return the next stage of the definition
+             */
+            WithImport withLocalKeyToImport(JsonWebKey key);
+        }
+
+        /**
+         * The stage of a key definition allowing to specify the key size.
+         */
+        interface WithKeySize {
+            /**
+             * Specifies the size of the key to create.
+             * @param size the size of the key in integer
+             * @return the next stage of the definition
+             */
+            WithCreate withKeySize(int size);
+        }
+
+        /**
+         * The stage of a key definition allowing to specify the allowed operations for the key.
+         */
+        interface WithKeyOperations {
+            /**
+             * Specifies the list of allowed key operations. By default all operations are allowed.
+             * @param keyOperations the list of JWK operations
+             * @return the next stage of the definition
+             */
+            WithCreate withKeyOperations(List<JsonWebKeyOperation> keyOperations);
+
+            /**
+             * Specifies the list of allowed key operations. By default all operations are allowed.
+             * @param keyOperations the list of JWK operations
+             * @return the next stage of the definition
+             */
+            WithCreate withKeyOperations(JsonWebKeyOperation... keyOperations);
+        }
+
+        /**
+         * The stage of a key definition allowing to specify whether to store the key in
+         * hardware security modules.
+         */
+        interface WithHsm {
+            /**
+             * Specifies whether to store the key in hardware security modules.
+             * @param isHsm store in Hsm if true
+             * @return the next stage of the definition
+             */
+            WithImport withHsm(boolean isHsm);
```

Comment:
What about two methods here, withHsm() and without(Hsm)? Or default for create to whatever the default is (I will guess isHsm == false which translate into one method with(Hsm)) and only have the two as part of the update def.